apispec/ext/marshmallow/openapi.py

```diff
@@ -330,6 +330,26 @@ def field2length(self, field, **kwargs):
                 attributes[max_attr] = validator.equal
         return attributes
 
+    def field2pattern(self, field, **kwargs):
+        """Return the dictionary of OpenAPI field attributes for a set of
+        :class:`Range <marshmallow.validators.Regexp>` validators.
+
+        :param Field field: A marshmallow field.
+        :rtype: dict
+        """
+        validators = [
+            validator for validator in field.validators
+            if hasattr(validator, 'regex')
+        ]
+
+        attributes = {}
+        for validator in validators:
+            # `SRE_Pattern` for < Python3.7
+            if type(validator.regex).__name__ in ('SRE_Pattern', 'Pattern'):
```

Comment:
Minor: This can be written a little more robustly with

```python
# compat.py

import re

RegexType = type(re.compile(''))
```

```python
from apispec.compat import RegexType

if isinstance(validator.regex, RegexType):
    #...
```

Reply:
I like this much better... done.